drivers/can/can_stm32.c

```diff
@@ -409,6 +410,50 @@ static int can_stm32_init(const struct device *dev)
 		return -EIO;
 	}
 
+	/* configure pinmux */
+	if (cfg->pinctrl_len != 0U) {
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32f1_pinctrl)
+		int remap;
+		/* Check that remap configuration is coherent across pins */
+		remap = stm32_dt_pinctrl_remap_check(cfg->pinctrl,
+						     cfg->pinctrl_len);
+		if (remap < 0) {
+			return remap;
+		}
+
+		/* A valid remapping configuration is provided */
+		/* Apply remapping before proceeding with pin configuration */
+		LL_APB2_GRP1_EnableClock(LL_APB2_GRP1_PERIPH_AFIO);
+
+		switch ((uint32_t)cfg->can) {
+#if defined(AFIO_MAPR_CAN_REMAP_REMAP1)
+		case DT_REG_ADDR(DT_NODELABEL(can1)):
+			if (remap == REMAP_1) {
+				LL_GPIO_AF_RemapPartial1_CAN1();
+			} else if (remap == REMAP_2) {
+				LL_GPIO_AF_RemapPartial2_CAN1();
+			} else if (remap == REMAP_3) {
+				LL_GPIO_AF_RemapPartial3_CAN1();
+			} else {
+				/* LL_GPIO_AF_DisableRemap_CAN1(); n/a */
+			}
```

Comment:
This should be something like rather
```suggestion
			if (remap == REMAP_1) {
			        /* PB8/PB9  */
				LL_GPIO_AF_RemapPartial2_CAN1();
			} else if (remap == REMAP_2) {
			       /* PD0/PD1 */
				LL_GPIO_AF_RemapPartial3_CAN1();
			} else {
				/* NO_REMAP: PA11/PA12 *//
				LL_GPIO_AF_RemapPartial1_CAN1();
			}
```

Reply:
Ok, thanks. Will fix that with next update.